### ISSUE TITLE
@extend-ing class is mistaken for placeholder

### ISSUE THREAD

**Issue by wardpenney** [brigade/scss-lint#480]
### Steps to recreate:

`scss-lint` version `0.34.0`

Have Scss:

```scss
.test__link {
  color: cornflowerblue;
}

.test__link--effect {
  @extend .test__link;
  text-decoration: underline;
}
```

Receive lint message "Name of placeholder .test__link--effect should be written in all lowercase letters with hyphens instead of underscores`. 

### Should do this

1. Linter should not automatically assume the `@extend` is being performed on a placeholder. It is [perfectly fine](http://sass-lang.com/documentation/file.SASS_REFERENCE.html#extend) to extend any valid declaration's selector.
2. When the error happens, the `hyphenated_lowercase` `CONVENTION` is enforced regardless of `PlaceholderInExtend` being `false`

Thanks for all the good work!

Edits:

* Updated to include underscores in example.
* Provided version

**Comment by sds** [comment]
Hey @wardpenney, thanks for the report.

I think the answer here is to remove the checking of placeholders from the `NameFormat` linter, as it doesn't really make sense (as has already been discussed in #472).

I've done that in b919a4228d1e885faacf03cbec5c0a2d2ac63a37.
